### ISSUE TITLE
Error in Automatic Differenciation for Functions with Powers of Zero

### ISSUE THREAD

**Issue by qbadolle** [google/jax#18640]
### Description

I am trying to vectorize a procedure, where I need to do the element-wise power of a large array before differentiating it.

If I have only one element, I can use this code which gives the expected output:

```
def func(x):
    x = x ** 0
    return x
jac = jacfwd(func)
jac(5.) # output = 0
```

However, when I try to vectorize this procedure, I get a different result.

```
def func(x):
    x = x ** jnp.array([[0]])
    return x
jac = jacfwd(func)
jac(5.) # output = 1
```

Is this expected? If yes, could you please explain the convention/reason? Thank you!

### What jax/jaxlib version are you using?

0.4.16 0.4.14

### Additional system info?

python 3.11.5 on macOS

**Comment by jakevdp** [comment]
Hi - thanks for the question! Using more recent jaxlib, I get the expected result. I think this issue was fixed by #17996, but you'll need to update to a more recent JAX release to get the fix – I think JAX v0.4.19 or newer should do it.

**Comment by qbadolle** [comment]
Indeed- both codes snippets give the same output with JAX v0.4.19.

Thank you!
